dojo/templates/dojo/snippets/engagement_list.html

```diff
@@ -202,21 +202,12 @@ <h4> {% if status == "open" %}Active{% elif status == "paused" %}Paused {% else
                     {% endif %}
                     <td class="text-center" class="nowrap">
                       <div class="align-top">
-                      {% if eng.test_set.all|length %}
+                      {% if eng.count_tests %}
                       <div class="dropdown">
-                        <a class="" href="{% url 'view_engagement' eng.id %}">{{ eng.test_set.all|length }}</a>
-                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">&nbsp;<i class="fa fa-ellipsis-v"></i>&nbsp;</a>
-                        <ul class="dropdown-menu">
-                          {% for test in eng.test_set.all %}
-                               <li>
-                                 <a class="" href="{% url 'view_test' test.id %}">
-                                   <i class="fa fa-list-alt"></i> {{test}}, {{test.updated}}
-                               </li>
-                            {% endfor %}
-                        </ul>
```

Comment:
not sure removing things to improve performance is the way to go

Reply:
I hadn't considered making it a "most recent test" menu - What do you think a reasonable limit is, n=10?

Reply:
maybe 25, but depends on if that still performs OK